compiler/ballerina-lang/src/main/java/io/ballerina/projects/repos/FileSystemCache.java

```diff
@@ -54,10 +55,18 @@ public class FileSystemCache extends CompilationCache {
     private final Path cacheDirPath;
     private Path birPath;
     private Path packageCacheDirPath;
+    private static Path buildProjectCacheDirPath;
```

Comment:
Why do we need this to be `static`?

Reply:
When testing the optimized executable, we have to cache the optimized thin JARs. Since these thin JARs are not reusable, we cache them inside the `<build project>/target/cache` directory. `buildProjectCacheDirPath` is used to find this `<build project>` directory. We update the static field when we set the cache for the build project and reuse it for optimized JARs.

I will try to refactor the logic to resolve the `/target` directory without using a static.

Reply:
After a discussion with @gayaldassanayake we used the `DefaultEnvironment` to share the `targetDirectory` cache of the `BUILD_PROJECT` with its imported `BALA_PROJECT`s. Now it is possible to access the `buildProjectCacheDirPath` without resorting to static fields. Changes can be found in 808143ae9036417e6d7e31366e3c12a9c546fc76.